### PR TITLE
Add better code check and pre-commit hook

### PR DESCRIPTION
**Pull request by while-false**
This PR changes how we do code style checks locally. It is the first step towards better code checking on our code spaces and on our Github Actions.

### CODE REVIEW

File: check-line-length [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+def print_line_check_result(result: LineCheckResult) -> None:
+    """Pretty print a line check result."""
+    for file_name, line_number, line in result:
+        print(f'{file_name}:{line_number+1}:{len(line)} {line.rstrip()}')
+
+
+if __name__ == '__main__':
+    import argparse
+    from multiprocessing import Pool
```
Comment (rjhuijsman) [inline check-line-length:81]: We usually keep all of our imports at the top of the file, no?

File: check-style [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# Get all files under version control.
+affected_files=$(git ls-tree --full-tree --name-only -r HEAD)
+
+# Cummulative status code for the script. Updated through 'run_check' when
+# running checks. A status code creater than 0 will indicate that one or more
+# checks failed.
+status_code=0
+
+# Run a code check command and update the cummulative error code.
+run_check() {
```
Comment (rjhuijsman) [inline check-style:22]: This file could use a comment explaining how to use it, most notably the expected arguments and what it does with them. It seems like we're supposed to pass a program to run?

File: pre-commit [hunk 3]
```diff
@@ -1,36 +1,6 @@
...
-
-case "$1" in
-  --help )
-    echo "Performs checks on files to be committed"
-    ;;
-  * )
-    run_checks
-    ;;
-esac
+# Run the check-style script from dev-tools as pre-commit hook.
```
Comment (rjhuijsman) [inline pre-commit:3]: Are you confident it's safe to drop the logic that finds the root of the git repository to run the checks there?

**Review by rjhuijsman — CHANGES_REQUESTED** [review]

**Comment by benh** [comment]
Hi all, let's drop this PR for now and revisit it in the future after we've focused on step one which is getting the precommit hook to be doing all of the things that the GitHub action is doing!

**Comment by while-false** [comment]
Closing in favour of https://github.com/3rdparty/dev-tools/pull/77

**Pull request closed without merging**